### PR TITLE
replace gulp-util dependency

### PR DESCRIPTION
**Pull request by ninbryan**
Hi,

This PR replaces deprecated dependency gulp-util

See:
- https://medium.com/gulpjs/gulp-util-ca3b1f9f9ac5
- https://github.com/gulpjs/gulp-util/issues/143

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -20,8 +20,9 @@
     "exec"
   ],
   "dependencies": {
-    "gulp-util": "~2.2.14",
-    "plexer": "0.0.1"
+    "plexer": "0.0.1",
+    "plugin-error": "0.1.2",
+    "vinyl": "2.1.0"
```
Comment (demurgos) [inline package.json]: I may be missing something but I think `vinyl` should in the `devDependencies` because it's only used in the tests.

**Review by demurgos — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (ninbryan) [reply package.json]: I will update dependencies shortly. What node version should it be for the travis.yml?

**Comment by ninbryan** [comment]
tests seems to not finish

**Comment by demurgos** [comment]
Thanks for updating the dependencies.

This is a bit odd. The test report is properly printed but it seems that mocha does not exit.

We usually defer CI issues to the maintainers, especially when it requires an update of the Node version (this is a breaking change so we prefer to let the maintainer do it).
If you can't solve it or the maintainer does not reply in a few day, I'll take a closer look at it.

**Comment by gRoussac** [comment]
Dear @demurgos,

Would you mind please merging this PR ?

Many thanks

**Comment by hparra** [comment]
Thanks all. This may be pre-existing bug.

It appears the sync/async tests are not terminating, due to newer behavior in Mocha: https://boneskull.com/mocha-v4-nears-release/#mochawontforceexit

We're seeing the new mocha due to upgraded Node. It's possible that older node/mocha combos also exhibit this behavior if `--no-exit` is passed in. This is worth testing to confirm.

See https://gist.github.com/boneskull/7fe75b63d613fa940db7ec990a5f5843

We can change `npm test` to force exit.

Or we can find what the underlying issue is:

```
Type: PROCESSWRAP
Error
    at AsyncHook.init (tmp/async-dump.js:12:62)
    at Process.emitInitNative (async_hooks.js:472:43)
    at new ChildProcess (internal/child_process.js:171:18)
    at Object.exports.spawn (child_process.js:498:15)
    at PassThrough.stream._transform (index.js:38:20)
    at PassThrough.Transform._read (_stream_transform.js:186:10)
    at PassThrough.Transform._write (_stream_transform.js:174:12)
    at doWrite (_stream_writable.js:387:12)
    at writeOrBuffer (_stream_writable.js:373:5)
    at PassThrough.Writable.write (_stream_writable.js:290:11)
    at Context.<anonymous> (tests/streamSync.js:77:11)


Type: PIPEWRAP
Error
    at AsyncHook.init (tmp/async-dump.js:12:62)
    at Pipe.emitInitNative (async_hooks.js:472:43)
    at internal/child_process.js:848:20
    at Array.reduce (<anonymous>)
    at _validateStdio (internal/child_process.js:825:17)
    at ChildProcess.spawn (internal/child_process.js:270:11)
    at Object.exports.spawn (child_process.js:502:9)
    at PassThrough.stream._transform (index.js:38:20)
    at PassThrough.Transform._read (_stream_transform.js:186:10)
    at PassThrough.Transform._write (_stream_transform.js:174:12)
    at doWrite (_stream_writable.js:387:12)
    at writeOrBuffer (_stream_writable.js:373:5)
    at PassThrough.Writable.write (_stream_writable.js:290:11)
    at Context.<anonymous> (tests/streamSync.js:77:11)
```

**Comment by demurgos** [comment]
@hparra 
Thanks for merging and looking into this issue.

By the way, if you look for maintainers you may be interested by the [Gulp community](https://github.com/gulp-community) organization. Repositories in this organization are still owned by the original authors (code style, license, npm publication) while granting access to members of the Gulp team to help with maintenance.

**Pull request merged**